### DISCUSSION TITLE
Pushing image with new tag to AWS ECR

### DISCUSSION THREAD

**Discussion by wwakas** [kreuzwerker/terraform-provider-docker#507] [Q&A]
Hi Guys,
I started my journey with terraform and I am facing an issue with pushing images to ECR using docker provider - lets began with my configuration:

```
#Create ECR repository for kafkaconnect image

resource "aws_ecr_repository" "kafkaconnect-aws" {
  name                 = "kafkaconnect-aws"
  image_tag_mutability = "IMMUTABLE"
}

#Build docker image and push to ECR

resource "docker_image" "kafkaconnect-aws-img" {
  name = "${aws_ecr_repository.kafkaconnect-aws.repository_url}:${var.kafka_connect_image_version}"
  build {
    context    = "./Dockerfiles"
    dockerfile = "Dockerfile"
  }
}

resource "docker_registry_image" "kafkaconnect-aws-ecr" {
  name          = docker_image.kafkaconnect-aws-img.name
  keep_remotely = true
}
```

so, I tested it and everything works well, except for one thing, when I try to change the docker image tag (${var.kafka_connect_image_version}) - the image with the changed tag is not pushed as a new image but only the tag is added to previous one, even if the repository is set to "tag immutability".

Any ideas about where am I wrong, or what should I change here to change the terraform behavior? 

Thanks in advance.

**Comment by Junkern** [comment]
As long as the image (and its contents) itself do not change, the `image_id` stays the same. And tags with the same `image_id` are shown in the same row in ECR. It does not mean that the previous tag is changed in anyway, only that those two tags have the same `image_id`.
